climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1165 @@
+    'WFsingle'.
+
+
+    Attributes:
+        date_end ((np.array): integer date corresponding to the proleptic
+            Gregorian ordinal, where January 1 of year 1 has ordinal 1
+            (ordinal format of datetime library))
+
+        n_fires (np.array): number of single fires in a fire season
+
+    """
+
+    days_thres_firms = 2
+    """ Minimum number of days to consider different fires """
+
+    clus_thres_firms = 15
+    """ Clustering factor which multiplies instrument resolution """
+
+    remove_minor_fires_firms = True
+    """ removes FIRMS fires below defined theshold of entries"""
+
+    minor_fire_thres_firms = 3
+    """ number of FIRMS entries required to be considered a fire """
+
+    prop_proba = 0.21
+    """ probability of fire propagation """
+
+    max_it_propa = 500000
+    """ maximum propgation iterations for a probabilistic fire """
+
+    def __init__(self, pool=None):
+        """Empty constructor. """
+        Hazard.__init__(self, HAZ_TYPE)
+        if pool:
+            self.pool = pool
+            LOGGER.info('Using %s CPUs.', self.pool.ncpus)
+        else:
+            self.pool = None
+
+    def set_hist_fire_FIRMS(self, csv_firms, centr_res_factor=1, centroids=None):
+        """ Parse FIRMS data and generate historical fires by temporal and spatial
+        clustering.
+
+        Parameters:
+            csv_firms: csv file of the FIRMS data (https://firms.modaps.eosdis.nasa.gov/download/)
+                or pd.DataFrame of FIRMS data
+            centr_res_factor (int, optional): resolution factor with respect to
+                the satellite data to use for centroids creation. Default: 1
+            centroids (Centroids, optional): centroids in degrees to map data
+        """
+        self.clear()
+
+        # read and initialize data
+        firms = self._clean_firms_csv(csv_firms)
+        # compute centroids
+        res_data = self._firms_resolution(firms)
+        if not centroids:
+            centroids = self._firms_centroids_creation(firms, res_data, centr_res_factor)
+        else:
+            if not centroids.coord.size:
+                centroids.set_meta_to_lat_lon()
+        res_centr = centroids._centroids_resolution(centroids)
+
+        # fire identification
+        while firms.iter_ev.any():
+            # Compute cons_id: consecutive fires in current iteration
+            self._firms_cons_days(firms)
+            # Compute clus_id: cluster identifier inside cons_id
+            self._firms_clustering(firms, res_data, self.clus_thres_firms)
+            # compute event_id
+            self._firms_fire(self.days_thres_firms, firms.cons_id.values, \
+                             firms.clus_id.values, firms.event_id.values, \
+                             firms.iter_ev.values, firms.datenum.values)
+            LOGGER.info('Remaining fires to identify: %s.', str(np.argwhere(\
+            firms.iter_ev.values).size))
```

Comment:
Is this needed for application or is this rather a debugging feature? If this typically takes very long, then you could for example provide info for every 10%, but not all fires. Or you could demote the logger to debug level, for example:
```suggestion
            LOGGER.debug('Remaining fires to identify: %s.', str(np.argwhere(\
            firms.iter_ev.values).size))
```

Reply:
I feel this is just good as it is.
On a large set you get roughly 4-8 updates from that which I personally like if you're computer is busy for ~10 minutes